src/test/scala/eu/proteus/solma/moments/WeightedMomentsEstimatorITSuite.scala

```diff
@@ -0,0 +1,148 @@
+
+import breeze.linalg
+import eu.proteus.annotations.Proteus
+import eu.proteus.solma.utils.FlinkTestBase
+import org.apache.flink.ml.math.Breeze._
+import org.apache.flink.ml.math.{DenseVector, Vector}
+import org.apache.flink.streaming.api.scala._
+import org.apache.flink.contrib.streaming.scala.utils._
+import org.scalatest.{FlatSpec, Matchers}
+import eu.proteus.solma.events.StreamEvent2
+
+@Proteus
+class WeightedMomentsEstimatorITSuite
+  extends FlatSpec
+    with Matchers
+    with FlinkTestBase {
+
+  behavior of "Flink's Weighted Moments Estimator"
+
+  import WeightedMomentsEstimator._
+  import WeightedMomentsEstimatorITSuite._
+
+  it should "estimate the mean and var of a stream" in {
+
+    val result = weightedMeanAndVariance(data,3)
+   val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setParallelism(3)
+    env.setMaxParallelism(3)
+
+    val stream = env.fromCollection(data)
+
+    val estimator = WeightedMomentsEstimator()
+      .setFeaturesCount(3)
+      .enableAggregation(true)
+    val it: scala.Iterator[WeightedMoments] = estimator.transform(stream).map(x=>x._2).collect()
+    val eps = 1E-5
+    while (it.hasNext) {
+      val elem = it.next
+      if (elem.counters(0) == 9) {
+        elem.mean(0) should be (result.weightedmean(0) +- eps)
+        elem.mean(1) should be (result.weightedmean(1) +- eps)
+        elem.mean(2) should be (result.weightedmean(2) +- eps)
+        elem.variance(0) should be (result.weightedvar(0) +- eps)
+        elem.variance(1) should be (result.weightedvar(1) +- eps)
+        elem.variance(2) should be (result.weightedvar(2) +- eps)
+      }
+    }
+  }
+
+   it should "estimate the mean and var of a stream on a features subset" in {
+
+
+
+     val result = weightedMeanAndVariance(data2,3)
+
+     val env = StreamExecutionEnvironment.getExecutionEnvironment
+     env.setParallelism(2)
+     env.setMaxParallelism(2)
+
+     val stream = env.fromCollection(data2)
+
+     val estimator = WeightedMomentsEstimator()
+       .setFeaturesCount(3)
+       .enableAggregation(true)
+
+     val it: scala.Iterator[WeightedMoments] = estimator.transform(stream).map(x=>x._2).collect()
+     val eps = 1E-5
+     while (it.hasNext) {
+       val elem = it.next
+       if ((elem.counters(0) == 3)&&(elem.counters(2) == 2)) {
+          elem.mean(0) should be (result.weightedmean(0) +- eps)
+         elem.mean(1) should be (result.weightedmean(1) +- eps)
+         elem.mean(2) should be (result.weightedmean(2) +- eps)
+        elem.variance(0) should be (result.weightedvar(0) +- eps)
+         elem.variance(1) should be (result.weightedvar(1) +- eps)
+         if (!java.lang.Double.isNaN(elem.variance(2))) {
+           elem.variance(2) should be (result.weightedvar(2) +- eps)
+         }
+       }
+     }
+   }
+
+}
+
+case class WeightedSample (
+                    s: IndexedSeq[Int],
+                    w:DenseVector,
+                    d: DenseVector) extends StreamEvent2 {
```

Comment:
I do not see this class in the unified view of commits, meaning this class does not exist, hence, you did not compile your changes.  Please, always make sure to commit tested code.